src/libraries/Microsoft.PowerFx.Core/Public/Values/FormulaValueNew.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.Public.Types;
+
+namespace Microsoft.PowerFx.Core.Public.Values
+{
+    // Basc new operators. 
+    public partial class FormulaValue
+    {
+        #region Host Utility API
+
+        // Host utility creation methods, listed here for discoverability.
+        // NOT FOR USE IN THE INTERPRETER! When creating new instances in
+        // the interpreter, call the constructor directly and pass in the
+        // IR context from the IR node.
+        public static NumberValue New(double number)
+        {
+            return new NumberValue(IRContext.NotInSource(FormulaType.Number), number);
+        }
+
+        public static FormulaValue New(double? number)
+        {
+            if (number.HasValue)
+            {
+                return New(number.Value);
+            }
+
+            return new BlankValue(IRContext.NotInSource(FormulaType.Number));
+        }
+
+        public static NumberValue New(decimal number)
+        {
+            // $$$ Is this safe? or loss in precision?
+            return new NumberValue(IRContext.NotInSource(FormulaType.Number), (double)number);
+        }
+
+        public static NumberValue New(long number)
+        {
+            // $$$ Is this safe? or loss in precision?
+            return new NumberValue(IRContext.NotInSource(FormulaType.Number), (double)number);
+        }
+
+        public static NumberValue New(int number)
+        {
+            return new NumberValue(IRContext.NotInSource(FormulaType.Number), number);
+        }
+
+        public static NumberValue New(float number)
+        {
+            return new NumberValue(IRContext.NotInSource(FormulaType.Number), number);
+        }
+
+        public static GuidValue New(Guid guid)
+        {
+            return new GuidValue(IRContext.NotInSource(FormulaType.Guid), guid);
+        }
+
+        public static StringValue New(string value)
+        {
+            return new StringValue(IRContext.NotInSource(FormulaType.String), value);
```

Comment:
>StringValue [](http://example.com/codeflow?start=23&length=11)

Check if value is null, and return BlankValue if that is the case

Reply:
This returns a StringValue, so it doesn't allow Blank. 
I'll just add the explicit null check.